src/routes/safe/container/LoadStore/index.jsx

```diff
@@ -0,0 +1,38 @@
+// @flow
+import { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import loadAddressBookFromStorage from '~/logic/addressBook/store/actions/loadAddressBookFromStorage'
+import addViewedSafe from '~/logic/currentSession/store/actions/addViewedSafe'
+import fetchLatestMasterContractVersion from '~/routes/safe/store/actions/fetchLatestMasterContractVersion'
+import fetchSafe from '~/routes/safe/store/actions/fetchSafe'
+import fetchTransactions from '~/routes/safe/store/actions/fetchTransactions'
+import { safeParamAddressFromStateSelector } from '~/routes/safe/store/selectors'
+
+type Props = {
+  setSafeLoaded: Function,
+}
+
+const LoadStore = (props: Props) => {
```

Comment:
I don't really like the idea of putting logic with no ui behind a component. Is there a particular reason why you decided to do that?

Reply:
I like the idea. https://alligator.io/react/smart-dumb-components/#what-makes-a-component-smart-or-dumb

It has several benefits

Reply:
I don't understand. I know what smart/dumb components are. This one doesn't really fit into this idea because:
> Smart components can fetch, capture changes and **pass down application data**.

 Why this component cant be a hook?

Reply:
you right, a hook would be better. perhaos a generic hook like this one https://scotch.io/tutorials/create-a-custom-usefetch-react-hook

Reply:
I made `fetchTokens` and `CheckForUpdates` hooks, it's not posible to do the same with `LoadStore`, because it's outside the provider context, is someone has an idea of how to solve it, feel free to give it and I'll implement that :)

Reply:
You can get the safeAddress not from redux but from "location" or something :)

Reply:
Yes but I need set a provider like this
```
 <Provider store={store}>
      <MuiThemeProvider theme={theme}>
        <ConnectedRouter history={history}>
          <PageFrame>
            <LoadStore setSafeLoaded={setSafeLoaded} />
            {wrapInSuspense(<AppRoutes safeLoaded={safeLoaded} />, <Loader />)}
          </PageFrame>
        </ConnectedRouter>
      </MuiThemeProvider>
    </Provider>
```

If I remove the component and use a hook it will be something like this:
```
const Root = () => {
  const [safeLoaded, setSafeLoaded] = useState(false)
  useLoadStore(setSafeLoaded)
  return (
    <Provider store={store}>
      <MuiThemeProvider theme={theme}>
        <ConnectedRouter history={history}>
          <PageFrame>
            {wrapInSuspense(<AppRoutes safeLoaded={safeLoaded} />, <Loader />)}
          </PageFrame>
        </ConnectedRouter>
      </MuiThemeProvider>
    </Provider>
  )
```

Reply:
I don't understand why this component is placed in `Root`. `safeLoaded` is used only in safe container. This just doesn't make sense to me - why on the top level-component we need to know if the safe is loaded and ready to be shown on the safe route? This seems like something to be put in the safe route logic